### DISCUSSION TITLE
Fatal error 404 (GET request)

### DISCUSSION THREAD

**Discussion by Sheld0n-Cooper** [kreait/firebase-php#558] [Q&A] [answered]
Hello, sorry it's me again ...

I think all of this is correct:

```
<?php

// Test PHP - Firebase

require __DIR__.'/vendor/autoload.php';

use Kreait\Firebase\Factory;

$factory = (new Factory)->withServiceAccount(__DIR__.'/secret/test-cb42f-f9d8f288b806.json');

$database = $factory->createDatabase();

$reference = $database->getReference('messages');

$value = $reference->getValue();

echo $value;
```

But I have an error: 

```
$ php index.php

Fatal error: Uncaught GuzzleHttp\Exception\ClientException: Client error: `GET https://test-cb42f.firebaseio.com/messages` resulted in a `404 Not Found` response:

 in C:\wamp64\www\zone42\testfirebase\vendor\guzzlehttp\guzzle\src\Exception\RequestException.php:113
Stack trace:
#0 C:\wamp64\www\zone42\testfirebase\vendor\guzzlehttp\guzzle\src\Middleware.php(69): GuzzleHttp\Exception\RequestException::create(Object(GuzzleHttp\Psr7\Request), Object(GuzzleHttp\Psr7\Response), NULL, Array, NULL)
#1 C:\wamp64\www\zone42\testfirebase\vendor\guzzlehttp\promises\src\Promise.php(204): GuzzleHttp\Middleware::GuzzleHttp\{closure}(Object(GuzzleHttp\Psr7\Response))
#2 C:\wamp64\www\zone42\testfirebase\vendor\guzzlehttp\promises\src\Promise.php(153): GuzzleHttp\Promise\Promise::callHandler(1, Object(GuzzleHttp\Psr7\Response), NULL)
#3 C:\wamp64\www\zone42\testfirebase\vendor\guzzlehttp\promises\src\TaskQueue.php(48): GuzzleHttp\Promise\Promise::GuzzleHttp\Promise\{closure}()
#4 C:\wamp64\www\zone42\testfirebase\vendor\guzzlehttp\promis in C:\wamp64\www\zone42\testfirebase\vendor\kreait\firebase-php\src\Firebase\Exception\DatabaseApiExceptionConverter.php on line 62
```

An idea ?
Thank you in advance.

**Comment by jeromegamez** [comment, marked as answer]
If you [search for "404"](https://github.com/kreait/firebase-php/discussions?discussions_q=404) in the discussions you will find the same question having been asked twice already 😅 

Please check on https://console.firebase.google.com/project/_/database/ what the name of your database is: You can find it right of the 🔗  symbol and it should look something like `https://<some-string>.firebaseio.com` (By default, the SDK will build this URL from the `project_id` field in your service account JSON file).

Once you have that URL, you can [configure it in the SDK with the `withDatabaseUri()` method of the `Factory` class](https://firebase-php.readthedocs.io/en/5.13.0/setup.html#custom-database-uri), so for example:

```php
$factory = (new Factory)
    ->withServiceAccount(../firebase.json)
    ->withDatabaseUri('https://my-database-name.firebaseio.com');

$database = $factory->createDatabase();
```

**Reply by Sheld0n-Cooper** [reply]
I'm sorry, I should have watched.
Thanks Jerome!

I did what you told me:

```
<?php

// Test PHP - Firebase

require __DIR__.'/vendor/autoload.php';

use Kreait\Firebase\Factory;

$factory = (new Factory)
    ->withServiceAccount(__DIR__.'/secret/test-cb42f-7cba9d5d4830.json')
    ->withDatabaseUri('https://test-cb42f.firebaseio.com');

$database = $factory->createDatabase();

$reference = $database->getReference('messages');

$value = $reference->getValue();

var_dump($value);
```

My name of my database is https://test-cb42f.firebaseio.com

(I am using Cloud Firestore)

I still have this error.

Here are some things that might help:
(Sorry this in French)

![img-firebase](https://user-images.githubusercontent.com/78118695/106211205-9cfb4b00-61c8-11eb-9a88-d1b28a2821f8.JPG)

![2](https://user-images.githubusercontent.com/78118695/106212434-eb114e00-61ca-11eb-99b9-059a9760fdfd.JPG)

**Reply by jeromegamez** [reply]
Ah, I thought you wanted to use the Realtime Database. For Firestore, you need another component 😅

https://firebase-php.readthedocs.io/en/5.14.1/cloud-firestore.html

**Reply by Sheld0n-Cooper** [reply]
Yes, yes I was wrong.
It is because of this video:
https://www.youtube.com/watch?v=z3aJO9Q10XU

I wanted to do the same, but it's not good.
Thanks Jerome.
